src/apify/_memory_storage/resource_clients/key_value_store.py

```diff
@@ -24,13 +27,28 @@
 )
 from ...consts import DEFAULT_API_PARAM_LIMIT, _StorageTypes
 from ...log import logger
-from ..file_storage_utils import _set_or_delete_key_value_store_record, _update_metadata
+from ..file_storage_utils import _update_metadata
 from .base_resource_client import BaseResourceClient
 
 if TYPE_CHECKING:
     from ..memory_storage_client import MemoryStorageClient
 
-DEFAULT_LOCAL_FILE_EXTENSION = 'bin'
+KeyValueStoreRecord = TypedDict('KeyValueStoreRecord', {'key': str, 'value': Any, 'contentType': Optional[str], 'filename': NotRequired[str]})
+
+
+def _filename_from_record(record: KeyValueStoreRecord) -> str:
+    if record.get('filename') is not None:
+        return record['filename']
+
+    content_type = record['contentType']
```

Comment:
```suggestion
    content_type = record.get('contentType')
```
not sure but, contentType can be undefined right?

Reply:
I think it should always be defined (but its value can possibly be `None`), but I guess better safe than sorry, I will update it